Tests/Admin/BreadcrumbsBuilderTest.php

```diff
@@ -0,0 +1,393 @@
+        );
+        $labelTranslatorStrategy->getLabel(
+            'dashboard',
+            'breadcrumb',
+            'link'
+        )->willReturn('Dashboard');
+        $admin->trans('Dashboard', array(), 'SonataAdminBundle')->willReturn(
+            'Tableau de bord'
+        );
+
+        $routeGenerator = $this->prophesize('Sonata\AdminBundle\Route\RouteGeneratorInterface');
+        $routeGenerator->generate('sonata_admin_dashboard')->willReturn('/dashboard');
+        $admin->getRouteGenerator()->willReturn($routeGenerator->reveal());
+        $menu->addChild('Tableau de bord', array('uri' => '/dashboard'))->willReturn(
+            $menu->reveal()
+        );
+        $labelTranslatorStrategy->getLabel(
+            'my_class_name_list',
+            'breadcrumb',
+            'link'
+        )->willReturn('My class');
+        if ($action == 'create') {
+            $labelTranslatorStrategy->getLabel(
+                'my_class_name_create',
+                'breadcrumb',
+                'link'
+            )->willReturn('create my object');
+            $admin->trans('create my object', array(), null)->willReturn('Créer mon objet')->shouldBeCalled();
+            $menu->addChild('Créer mon objet', array())->willReturn($menu);
+        }
+        $childAdmin = $this->prophesize('Sonata\AdminBundle\Admin\AbstractAdmin');
+
+        $admin->trans('My class', array(), null)->willReturn('Ma classe');
+        $admin->hasRoute('list')->willReturn(true);
+        $admin->isGranted('LIST')->willReturn(true);
+        $admin->generateUrl('list')->willReturn('/myadmin/list');
+        $admin->getCurrentChildAdmin()->willReturn(
+            $action == 'my_action' ? $childAdmin->reveal() : false
+        );
+        if ($action == 'list') {
+            $admin->isChild()->willReturn(true);
+            $menu->setUri(false)->shouldBeCalled();
+        }
+        $request = $this->prophesize('Symfony\Component\HttpFoundation\Request');
+        $request->get('slug')->willReturn('my-object');
+
+        $admin->getIdParameter()->willReturn('slug');
+        $admin->hasRoute('edit')->willReturn(false);
+        $admin->getRequest()->willReturn($request->reveal());
+        $admin->hasSubject()->willReturn(true);
+        $admin->getSubject()->willReturn('my subject');
+        $admin->toString('my subject')->willReturn('My subject');
+        $admin->getLabelTranslatorStrategy()->willReturn(
+            $labelTranslatorStrategy->reveal()
+        );
+        $admin->getClassnameLabel()->willReturn('my_class_name');
+
+        $menu->addChild('Ma classe', array(
+            'uri' => '/myadmin/list',
+        ))->willReturn($menu->reveal());
+        $menu->addChild('My subject')->willReturn($menu);
+        $menu->addChild('My subject', array('uri' => null))->willReturn($menu);
+
+        $breadcrumbsBuilder->buildBreadCrumbs($admin->reveal(), $action);
+    }
+}
+
+class DummySubject
```

Comment:
Wow that's strange, I already fixed that… some force push went wrong I 
guess.

Le 23/05/2016 15:12, Oskar Stark a écrit :
>
> In Tests/Admin/BreadcrumbsBuilderTest.php 
> <https://github.com/sonata-project/SonataAdminBundle/pull/3791#discussion_r64219170>:
>
> > +        $admin->getLabelTranslatorStrategy()->willReturn(
> > +            $labelTranslatorStrategy->reveal()
> > +        );
> > +        $admin->getClassnameLabel()->willReturn('my_class_name');
> > +
> > +        $menu->addChild('Ma classe', array(
> > +            'uri' => '/myadmin/list',
> > +        ))->willReturn($menu->reveal());
> > +        $menu->addChild('My subject')->willReturn($menu);
> > +        $menu->addChild('My subject', array('uri' => null))->willReturn($menu);
> > +
> > +        $breadcrumbsBuilder->buildBreadCrumbs($admin->reveal(), $action);
> > +    }
> > +}
> > +
> > +class DummySubject
>
> is this moved and the diff is wrong or is it still in this file?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/sonata-project/SonataAdminBundle/pull/3791/files/a0995099b8c3bcad19503ffa2da43d3b1a18d39f#r64219170>
>